### ISSUE TITLE
Genesis out of sync on Linux

### ISSUE THREAD

**Issue by undron** [neodix42/MyLocalTon#38]
Running MyLocalTon on Ubuntu 20.04.4 LTS leads to out of sync.
```
14:21:43.214 [MyLocalTon - node3 Status Monitor] INFO  org.ton.actions.MyLocalTon - node3 is not ready
14:21:43.214 [MyLocalTon - node2 Status Monitor] DEBUG o.t.e.liteclient.LiteClientParser - Error parsing lite-client's last command! Output: using liteserver 0 with addr [127.0.0.1:4443]
zerostate set to -1:526F1573282C41194EDF83D684B7C4505205885159013664E4DF16DC65B4698F:E297BEC613F2E8A87AF881277A241AD336FC4854111FC766216077A40E4614F7

14:21:43.214 [MyLocalTon - node2 Status Monitor] INFO  org.ton.actions.MyLocalTon - node2 is not ready
14:21:43.214 [MyLocalTon - node7 Status Monitor] DEBUG o.t.e.liteclient.LiteClientParser - Error parsing lite-client's last command! Output: using liteserver 0 with addr [127.0.0.1:4443]
zerostate set to -1:526F1573282C41194EDF83D684B7C4505205885159013664E4DF16DC65B4698F:E297BEC613F2E8A87AF881277A241AD336FC4854111FC766216077A40E4614F7

14:21:43.215 [MyLocalTon - node7 Status Monitor] INFO  org.ton.actions.MyLocalTon - node7 is not ready
14:21:43.215 [validator-engine-node5] INFO  o.t.e.v.ValidatorEngineExecutor - validator-engine-node5 stopped
14:21:43.215 [validator-engine-node5] DEBUG o.t.e.v.ValidatorEngineExecutor - validator exit output:  
14:21:43.215 [MyLocalTon - Dump Block 478] DEBUG o.t.e.liteclient.LiteClientParser - Error parsing lite-client's last command! Output: using liteserver 0 with addr [127.0.0.1:4443]
zerostate set to -1:526F1573282C41194EDF83D684B7C4505205885159013664E4DF16DC65B4698F:E297BEC613F2E8A87AF881277A241AD336FC4854111FC766216077A40E4614F7
```

Command to run:
`java -jar MyLocalTon-x86-64.jar debug  with-validators-6`
`/home/dockeruser1/.local/bin/ton-http-api --liteserver-config /home/dockeruser1/ton/ton-home/myLocalTon/genesis/db/my-ton-global.config.json --port 8081`
I'm eunning `ton-http-api` in separate process just to monitor results, running it within MyLocalTon produces the same results.

Versions:
Ubuntu 20.04.4 LTS
MyLocalTon v123
openjdk 17.0.12 2024-07-16

Logs: https://drive.google.com/drive/folders/1pfw2Pgvt8EZn18QkqtBV3SjONZuiGcnZ?usp=sharing

**Comment by neodix42** [comment]
Please try v124,.

**Comment by undron** [comment]
Still have the issue. Logs:
https://drive.google.com/drive/folders/1E8MJXVWDbwXQZV4dOJJpQYfBzXIAp1Ow?usp=sharing
https://drive.google.com/drive/folders/1bUb1R-PCgu-rXTA1qu5SSaIOaDIO-sfD?usp=sharing

**Comment by neodix42** [comment]
Please attach the logs here or send me on telegram @neodix.

**Comment by drucewihlis** [comment]
Hi there, using v124 dockerized and non-dockerized. In both cases same issue `Cannot parse null string` appears and then genesis goes out of sync.

Below I attach logs from `/myLocalTon/genesis/db/log` when I run the app non-dockerized with `java -jar MyLocalTon-x86-64-testnet.jar ton-http-api explorer nogui with-validators-6`:
[log_no_docker_cannot_parse_nul_str.zip](https://github.com/user-attachments/files/18059818/log_no_docker_cannot_parse_nul_str.zip)

**Comment by neodix42** [comment]
Hello guys,
First of all I would like to apologize that the quality of MyLocalTon having not the best times.
The reason for that is that portable binaries built with NixPkgs were always not very stable and in the meantime they became even more unstable. Desynchronization issues you experiece are caused due to this ugly error, that comes out during validation process:

```
[pid 18710] [time 1733730037] Signal: 11
------- Stack Backtrace -------
[0x683f91]
[0x6842a1]
[0x681038]
[0x17e9ef0]
[0xd1224c]
[0xd12ab0]
[0xd1343b]
[0xd15ea3]
[0xca2ddd]
[0x156b37c]
[0x1563891]
[0x155e9f3]
[0x14629a7]
[0x12ae548]
[0xd51a0c]
[0xd5bf91]
[0xd62466]
[0xd4ba7e]
[0xd2cca9]
[0x6adef7]
[0x6ae14d]
[0x6bf943]
[0x6b194b]
[0x5a7487]
[0x180d66e]
[0x1853f90]
-------------------------------
```
This error appears in both dockerized and non-dockerized MyLocalTon versions. 
I am fully aware of the problem and thought that the v124 was the good one. Actaully it improved stability very much on MacOS systems. 

I will continue investigating the reasons and will put even more efforts to make it back stable again. 

And, exactly for docker-lovers, I have already started to work on Docker-Only MyLocalTon version. This one gonna be very stable, since will be build based on official TON docker image.

**Comment by dkatzan** [comment]
@neodix42 can you please make the docker image cross compiled?

**Comment by neodix42** [comment]
> @neodix42 can you please make the docker image cross compiled?

yes, the official image supports both arm64 and amd64, and MyLocalTon docker image will be built in the same way.

**Comment by dkatzan** [comment]
Hi, looks great
will try to spin it up this week (we need some extra work on our side, since our current docker is actually also running opentonapi and some small fastapi server with some management operation for our tests, so might take a bit to actually integrate it)
